### ISSUE TITLE
macOS build does not support HDF5 SWMR

### ISSUE THREAD

**Issue by tschoonj** [vasole/pymca#59]
Hi Armando,

One of our users is complaining as they cannot open our SWMR Nexus files with the latest macOS build. I checked and it ships with HDF5 1.8.x. Would it be possible to generate a macOS build with the latest HDF5 libraries?

Many thanks in advance!

**Comment by vasole** [comment]
Sure, but I am afraid that one is going to take longer.

**Comment by tschoonj** [comment]
Thanks!

**Comment by tschoonj** [comment]
Hi Armando,

Since this was rather urgent, I decided to hack your latest macOS release and move in the latest version of HDF5 (1.10.0-patch1) as well as the latest h5py, along with some fooling around with install_name_tool to update the references to the dylib paths. This appears to work fine. In case anyone needs this, the modified macOS release can be found [here](https://www.dropbox.com/s/su1i2k3j5lam9q6/PyMca5.1.4-SWMR.zip?dl=0)

**Comment by vasole** [comment]
Hi Tom,

Glad you found a temporary solution.

Did you just dropped the new files into the supplied app? I would like to ship your modified code, but unless you used python from python org and a 10.7 Mac I cannot afford possible incompatibilities.

**Comment by tschoonj** [comment]
Hi Armando,

No it was more than just dropping the files: I compiled the HDF5 library with Homebrew, and built h5py from source with pip (no binary was used) against the same version of Numpy (1.9.2, also installed with pip from source) that you used in the bundle. The python I used was 2.7.13, installed with Homebrew. I had to use `install_name_tool` to ensure run-time link dependencies for the h5py binary modules would work, since by default they will look only in the Homebrew folder for the HDF5 library.

This all happened on my iMac running Sierra, so there's no guarantee it will work on older versions, but perhaps it will.

You are very generous setting 10.7 as minimum supported version by the way since it is no longer supported by Apple: it no longer receives security updates for example, which makes this version quite vulnerable.

Feel free to use or distribute it as you see fit. So far it has been tested on two machines and it works as expected on both, with the additional benefit of SWMR HDF5 support 😄
